cmd/govim/util.go

```diff
@@ -44,3 +50,82 @@ func (v *vimstate) cursorPos() (b *types.Buffer, p types.Point, err error) {
 	p, err = types.PointFromVim(b, pos.Line, pos.Col)
 	return
 }
+
+func (v *vimstate) populateQuickfix(locs []protocol.Location) {
+	// must be non-nil
+	qf := []quickfixEntry{}
+
+	for _, loc := range locs {
+		var buf *types.Buffer
+		for _, b := range v.buffers {
+			if b.Loaded && b.URI() == span.URI(loc.URI) {
+				buf = b
+			}
+		}
+		fn := span.URI(loc.URI).Filename()
+		if buf == nil {
+			byts, err := ioutil.ReadFile(fn)
+			if err != nil {
+				v.Logf("populateQuickfix: failed to read contents of %v: %v", fn, err)
+				continue
+			}
+			// create a temp buffer
+			buf = types.NewBuffer(-1, fn, byts, false)
+		}
+		// make fn relative for reporting purposes
+		fn, err := filepath.Rel(v.workingDirectory, fn)
+		if err != nil {
+			v.Logf("populateQuickfix: failed to call filepath.Rel(%q, %q): %v", v.workingDirectory, fn, err)
+			continue
+		}
+		p, err := types.PointFromPosition(buf, loc.Range.Start)
+		if err != nil {
+			v.Logf("popularQuickfix: failed to resolve position: %v", err)
+			continue
+		}
+		line, err := buf.Line(p.Line())
+		if err != nil {
+			v.Logf("popularQuickfix: location invalid in buffer: %v", err)
+			continue
+		}
+		qf = append(qf, quickfixEntry{
+			Filename: fn,
+			Lnum:     p.Line(),
+			Col:      p.Col(),
+			Text:     line,
+		})
+	}
+
+	var toSort []quickfixEntry
+
+	// For references there is a special case where the first
+	// value returned by gopls is the definition.
+	fv := qf[0].Text
+	if strings.Contains(fv, ":=") || strings.Contains(fv, "var") {
```

Comment:
This is a bit hacky :) The caller of this method knows whether we are in references mode or not. So you could simply have a boolean parameter that determines whether you sort the entire slice or `[1:]`

Reply:
Ah, good to know - I'll fix this up 😄